redis/tests/test_cluster_async.rs

```diff
@@ -447,6 +447,233 @@ fn test_async_cluster_move_error_when_new_node_is_added() {
     assert_eq!(value, Ok(Some(123)));
 }
 
+fn test_async_cluster_move_error_refresh_topology(
```

Comment:
this function is almost 100 lines long, and reading its name doesn't explain what it is testing. this makes it very hard to verify the test's correctness.